src/Tools/BuildBoss/CompilerNuGetCheckerUtil.cs

```diff
@@ -75,18 +76,25 @@ private bool CheckDesktop(TextWriter textWriter)
             // inside of desktop MSBuild. Even though they are in our output directories they should
             // not be a part of our deployment
             // need to be 
-            var unneededDllFileNames = new[]
-            {
+            dllRelativeNames = FilterRelativeFileNames(
+                dllRelativeNames,
                 "Microsoft.Build.dll",
                 "Microsoft.Build.Framework.dll",
                 "Microsoft.Build.Tasks.Core.dll",
-                "Microsoft.Build.Utilities.Core.dll",
-            };
-            dllRelativeNames = dllRelativeNames
-                .Where(x => !unneededDllFileNames.Contains(x, PathComparer))
-                .ToList();
+                "Microsoft.Build.Utilities.Core.dll").ToList();
+
+            // The SWR files don't ship the native dia libraries as they are shipped by a different 
+            // team in Visual Studio.
+            var swrRelativeFileNames = FilterRelativeFileNames(
+                dllRelativeNames,
+                "Microsoft.DiaSymReader.Native.amd64.dll",
+                "Microsoft.DiaSymReader.Native.x86.dll").ToList();
 
-            allGood &= VerifySwrFile(textWriter, dllRelativeNames);
+            // The Microsoft.CodeAnalysis.Compilers.swr file is used in part to ensure NGEN is run on the set of 
+            // facades / implementation DLLs the compiler depends on. This set of DLLs is the same as what is 
+            // included in our NuGet package. Need to make sure all the necessary managed DLLs are included here.
+            // TODO: fix this
```

Comment:
TODO comment seems...unclear. What needs fixing?

Reply:
Narrator: he saw that comment and fixed the issue but left the comment because ...